src/hypertable.c

```diff
@@ -561,19 +569,40 @@ hypertable_get_by_id(int32 hypertable_id)
 								   &ht,
 								   1,
 								   AccessShareLock,
-								   false);
+								   false,
+								   CurrentMemoryContext);
 	return ht;
 }
 
+static ChunkStoreEntry *
+hypertable_chunk_store_add(Hypertable *h, Chunk *chunk)
+{
+	ChunkStoreEntry *cse;
+	MemoryContext old_mcxt,
+				chunk_mcxt;
+
+	chunk_mcxt = AllocSetContextCreate(subspace_store_mcxt(h->chunk_cache),
+									   "chunk cache entry memory context",
+									   ALLOCSET_SMALL_SIZES);
+
+	/* Add the chunk to the subspace store */
+	old_mcxt = MemoryContextSwitchTo(chunk_mcxt);
+	cse = palloc(sizeof(ChunkStoreEntry));
+	cse->mcxt = chunk_mcxt;
+	cse->chunk = chunk;
```

Comment:
Shouldn't this need a chunk_copy as it did before?

Reply:
I thought this PR would get rid of the code that does chunk_find on another memory context but then copies it over, however as far as I can tell that's not what's happening here and the copy would still be necessary. So I don't see the point here.

Reply:
I don't see how it would be possible to get rid of the copying since `chunk_find` searches for a chunk with N slices (in N dimensions), and a scan might involve some slices from other chunks. Therefore you do not know the exact N slices until all are found for the specific chunk.

Reply:
Regarding `chunk_copy`, I think you are right that it is needed. The reason it is not there now is that I broke this code out from another PR that had a relid chunk cache where the chunk lived. Will fix.